### PR TITLE
Improved Botany swabs

### PR DESCRIPTION
**Pull request by Sirionaut**
## About the PR
Made a distinct pollen swab so I could change the description in hopes of making the functionality more clear.
~~Will show the last plant that was swabbed.~~

## Why / Balance
After seeing all those discussions under a certain someone's youtube videos about the topic, I thought I try to make it more obvious in game how they work. No more misleading "single use".

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame
swab

## Breaking changes

**Changelog**

doesn't need a cl I think?

### CODE REVIEW

**Comment by lzk228** [comment]
i personally dont like it, like you should remember or mark your swabs with labeler
when plant analyzer will be in game its better to allow scan swabs with it

**Comment by Krunklehorn** [comment]
Plant analyzer is not here yet and this change will be trivial to remove when it comes.

Swabs are unintuitive. You really only understand them by hearsay from experienced Botanists, and even then, the behavior is not easy to articulate in words.

This PR helps avoid a common pitfall for new Botanists.

**Comment by Partmedia** [comment]
"Single use" is a hint that something unexpected may happen if you use it multiple times.

I don't like that the swab displays the identity of the pollen on it. Examine text is designed to convey easily discernible information. You cannot examine a player, for example, and have it say "they are a traitor". Similarly, I don't think swabs should indicate what kind of pollen is on it.

**Comment by Keikiru** [comment]
> "Single use" is a hint that something unexpected may happen if you use it multiple times.
> 
> I don't like that the swab displays the identity of the pollen on it. Examine text is designed to convey easily discernible information. You cannot examine a player, for example, and have it say "they are a traitor". Similarly, I don't think swabs should indicate what kind of pollen is on it.

Comparing this to the traitor role is an extreme example. Something less overly egregious would be comparing it to examining a player to see if they have been wounded. A wounded player's state changes from an unwounded one, which is often (but not always, depending on damage types) reflected in a visible sprite change and additional examination text.

This is a simple quality of life feature. It primarily helps forgetful or distracted players, and is by no means a tremendous harm to the integrity of the game or botany as a department.
It doesn't even show any detailed information like what sort of genes or mutations might be on the swab.

As Krunklehorn points out:

>Plant analyzer is not here yet and this change will be trivial to remove when it comes.

This is a non-issue until the plant analyzer is, if ever, added. Holding up a quality of life feature for a potential future hypothethical is not reasonable.

>Swabs are unintuitive. You really only understand them by hearsay from experienced Botanists, and even then, the behavior is not easy to articulate in words.
>
>This PR helps avoid a common pitfall for new Botanists.

This goes some way to help alleviate the occult nature of swabbing.

>i personally dont like it, like you should remember or mark your swabs with labeler

Swabs are used two entire times. Once to gather from a plant, and a second time to apply to another plant. To assume a player wrangle the clunky labeler UI for each and every swab is ridiculous when that added complexity and frustration can be easily partially alleviated with this change.

**Comment by Krunklehorn** [comment]
#24904 

This is what I mean, it's completely unintuitive. People don't even understand it when they take the time to _try_ and understand it.

At the very least the description needs to change.

**Comment by GreyMario** [comment]
I'm in the "this is wholly unnecessary" camp on this one.

Unless you'd like to argue that Medical should be able to use their naked eye to identify what chemicals are inside of a patient?

**Comment by Varen** [comment]
If you don't already know how it works because someone explained it to you, or you watched info about it elsewhere, the swabs are just incredibly unintuitive to use.
This helps that. It's good for new players who just started on the role. For everyone else, you're not even looking at it anyway so why complain? 
Let new players have their training wheels on one of the most weirdly confusing roles in the game.

**Comment by Keikiru** [comment]
> I'm in the "this is wholly unnecessary" camp on this one.
> 
> Unless you'd like to argue that Medical should be able to use their naked eye to identify what chemicals are inside of a patient?

Please read my entire comment. I already state this provides far less information than a full gene readout. It's far closer to seeing of a player has been burned, stabbed, bruished and such. You are making a false equivalence.

**Comment by K-Dynamic** [comment]
> "Single use" is a hint that something unexpected may happen if you use it multiple times.
> 
> I don't like that the swab displays the identity of the pollen on it. Examine text is designed to convey easily discernible information. You cannot examine a player, for example, and have it say "they are a traitor". Similarly, I don't think swabs should indicate what kind of pollen is on it.

I'm on the fence because swabs shouldn't be readable without a device or machine but they're also really unfriendly to new players atm

**Comment by gordod3** [comment]
PR fair enough, but better **make** them "single use only" for sure because it make more interaction between Botany and Medbay there they can print more swabs or Cargo where can buy medkits with swabs. Reusable swabs sounds wrong and cursed, no sense hiding and called it now "Pollen swab".

**Comment by Partmedia** [comment]
A typical person cannot look at swab and 1) tell if there's pollen on it, nevermind 2) what type of pollen is on it. This is why this is different from the medical examine, because if you just look at someone a reasonable person can tell if they're bleeding or not.

I'd be willing to compromise on adding this new botany swab with this description if the pollen identification examine is removed.

**Comment by gordod3** [comment]
Ah ye I forgot about it! Write what plant was in last swab is wrong because botanist can't properly say what kind dust it is without equipment(like analazyer in theory with function analyze gens on swab(Don't know can analyze make it for sure, but why not)) and your watermelon can be already gen changed so you can't know be certinaly what kind "watermelon" dust you have on swab.

**Comment by Sirionaut** [comment]
As lzk said at the beginning, one can achieve the same result by just remembering what plant was last swabbed, or use a labeler if you want. (plenty of botanists like to... "method act" while growing space weed, so remembering is hard...)

So why not make life easier for players? This is the simplest QoL add for botany that will prevent a large chunk of new botanists from learning to swab wrong and prevent confusion.

Just to be clear, this **only shows the display name** of a plant, nothing more, the same you see on tray examine.
If used on an apple tree whose apples contain flour, it will still say apple tree.

The guidebook has become much better for new botanists compared to when I started, but having training wheels for new players in this regard is still a good idea, I am sure.

I would challenge anyone to explain how swabs work in game to a new player without the use of ooc. Because is just started to lie and say to throw them away after one use because they never believe me when the description contradicts me.

I did hope for more concrete feedback for the text itself. 
>Transfers pollen between plants, can lead to addition or removal of plant traits.

Someone got a shot and better sentence?

Maybe change the lower text to "This swab was last used on a XYZ plant". Not using the word pollen so its even more clear it just shows what you swabbed as a memory help, nothing more.

If we remove the last swabbed plant part, should it still be a new prototype or just change the old item desc since viro is far away?

**Comment by Partmedia** [comment]
> So why not make life easier for players? This is the simplest QoL add for botany that will prevent a large chunk of new botanists from learning to swab wrong and prevent confusion.
> 
> Just to be clear, this **only shows the display name** of a plant, nothing more, the same you see on tray examine. If used on an apple tree whose apples contain flour, it will still say apple tree.

Because examine text should not be used to convey information that is not easily discernible to players. See above examples on "examine to see what liquids are in your stomach, examine to see if you are traitor or not."

> I would challenge anyone to explain how swabs work in game to a new player without the use of ooc. Because is just started to lie and say to throw them away after one use because they never believe me when the description contradicts me.

"The genes that convey traits are in the pollen. You can pick up pollen with a swab and transfer it to another plant for a chance to exchange traits. You should not reuse a swab because then it's easy to lose track of what pollen you're transferring from, but if you think you're really smart you can go ahead and do it anyway."

> > Transfers pollen between plants, can lead to addition or removal of plant traits.

This is a run-on sentence. You need a clause to join the second clause to the first, e.g. "which".

**Comment by Partmedia** [comment]
> Are we okay with plant analyzer reading swabs for pollen information?

If a plant analyzer exists and works on plants, then I think it's reasonable for it to work on a swab with pollen.

**Comment by Sirionaut** [comment]
> Because examine text should not be used to convey information that is not easily discernible to players.

It is easily discernible information, since the player is the one who just used the swab? It's just meant as an "automated" labeling that makes it obvious what is happening, not revealing new information. 

I struggle to follow what is too much information and what isn't.
I love the new power monitoring console, but it offers a vast amount of previously hidden information that makes certain antag strats much harder, that has actual gameplay/balance impact unlike this here. 
Do we have guidelines on how to approach the topic of how much information to display? Seems I got the wrong ideas how others want this game to work.

>"The genes that convey traits are in the pollen. You can pick up pollen with a swab and transfer it to another plant for a chance to exchange traits. You should not reuse a swab because then it's easy to lose track of what pollen you're transferring from, but if you think you're really smart you can go ahead and do it anyway."

I will steal this.

So to summarize: add a "which" in the sentence, remove the plant.DisplayName, but keep it as a separate from the disease swab?

**Comment by Partmedia** [comment]
> It is easily discernible information, since the player is the one who just used the swab? It's just meant as an "automated" labeling that makes it obvious what is happening, not revealing new information.
> 
> I struggle to follow what is too much information and what isn't. I love the new power monitoring console, but it offers a vast amount of previously hidden information that makes certain antag strats much harder, that has actual gameplay/balance impact unlike this here. Do we have guidelines on how to approach the topic of how much information to display? Seems I got the wrong ideas how others want this game to work.

Unfortunately we don't have an "examinable design doc" anywhere. It is up to maintainers to enforce consistency in what gets merged.

> So to summarize: add a "which" in the sentence, remove the plant.DisplayName, but keep it as a separate item from the disease swab?

Yes. In short, this will be a YAML-only change.

**Comment by Sirionaut** [comment]
made the requested changes.
Reminder there is an overlapping PR to make a pollen swab dispenser.

File: Content.Server/Botany/Systems/BotanySwabSystem.cs
Comment (Partmedia) [inline Content.Server/Botany/Systems/BotanySwabSystem.cs:33]: Why not use the existing FTL strings?

Comment (Sirionaut) [reply Content.Server/Botany/Systems/BotanySwabSystem.cs:33]: New item new FTL string made sense in my head. Else I would have just changed the describtion and not make a new item?
Not much of a medical player so I have no idea if they are mapped outside of botany, which could lead to confusion, was my thinking.

**Review by Partmedia — CHANGES_REQUESTED** [review]
Given that:

- The sterile swab is otherwise unused,
- Sterile swabs are not functionally different from botany swabs, and
- They share the same guidebook entries

I think it would just be best if there was no new entity, and instead the name and/or description of the sterile swab change.

**Comment by Partmedia** [comment]
Based on the feedback from the linked issue, I think it would be more clear if swabs came in a wrapped "sterile" package that needs to be unwrapped before use, like chocolate bars.

Alternatively, we can just adjust the name and/or description for the existing item. Let's not create an additional item just to change the name/description.

**Pull request closed without merging**
